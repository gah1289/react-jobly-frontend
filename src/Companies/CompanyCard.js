import React from 'react';
import JoblyApi from '../api';
import './Companies.css';
import { Card, CardBody, CardTitle, CardSubtitle, CardText, Button, CardImg } from 'reactstrap';

import { useNavigate } from 'react-router-dom';

import logo1 from './logos/logo1.png';
import logo2 from './logos/logo2.png';
import logo3 from './logos/logo3.png';
import logo4 from './logos/logo4.png';
import noLogo from './logos/no-logo.png';

function CompanyCard({ company }) {
	const navigate = useNavigate();

	const getLogo = (logoUrl) => {
		let logo;
		let logoNum;
		if (logoUrl) {
			logoNum = logoUrl[11];
			if (logoNum === '1') {
				logo = logo1;
			}
			if (logoNum === '2') {
				logo = logo2;
			}
			if (logoNum === '3') {
				logo = logo3;
			}
			if (logoNum === '4') {
				logo = logo4;
			}
		}
		else logo = noLogo;
		return logo;
	};

	const getCompanyJobs = async (co) => {
		let jobsFromApi = await JoblyApi.getJobs();
		let filteredJobs = jobsFromApi.filter((job) => job.companyHandle == co);

		return navigate('/jobs', { state: { props: filteredJobs } });
	};

	return (
		<Card style={{ minWidth: '30vw' }} className="company-card">
			<CardImg
				style={{ height: '150px', width: '200px', margin: 'auto' }}
				alt={company.name}
				src={getLogo(company.logoUrl)}
				top
				width="100%"
			/>
			<CardBody>
				<CardTitle tag="h5">{company.name}</CardTitle>
				<CardSubtitle className="mb-2 text-muted" tag="h6">
					Size: {company.numEmployees} Employees
				</CardSubtitle>
				<CardText>{company.description}</CardText>

				<Button
					onClick={async () => {
						await getCompanyJobs(company.handle);
					}}
				>
					Jobs
				</Button>
			</CardBody>
		</Card>
	);
}

export default CompanyCard;
