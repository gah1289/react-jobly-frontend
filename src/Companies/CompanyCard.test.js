import { render, screen } from '@testing-library/react';
import CompanyCard from './CompanyCard';
import ItemContext from '../ItemContext';
import { MemoryRouter } from 'react-router-dom';

describe('Loads Company Card', function() {
	let isLoggedIn = true;

	let isLoading = false;

	let user = {
		username  : 'testUser',
		firstName : 'Test',
		lastName  : 'User',
		email     : 'test@user.com',
		password  : 'password'
	};

	let company = {
		handle       : 'anderson-arias-morrow',
		name         : 'Anderson, Arias and Morrow',
		description  :
			'Somebody program how I. Face give away discussion view act inside. Your official relationship administration here.',
		numEmployees : 245,
		logoUrl      : '/logos/logo3.png'
	};

	let CompanyCardComponent = (
		<MemoryRouter>
			<ItemContext.Provider value={{ isLoading, isLoggedIn, user }}>
				{' '}
				<CompanyCard company={company} />{' '}
			</ItemContext.Provider>
		</MemoryRouter>
	);

	it('renders without crashing', function() {
		render(CompanyCardComponent);
	});

	it('shows card with company information', function() {
		const { asFragment } = render(CompanyCardComponent);
		expect(asFragment()).toMatchSnapshot();
		const cardTitle = screen.queryByText(/Anderson/i);
		expect(cardTitle).toBeInTheDocument();
	});
});
