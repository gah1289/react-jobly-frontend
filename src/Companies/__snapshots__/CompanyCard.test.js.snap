// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loads Company Card shows card with company information 1`] = `
<DocumentFragment>
   
  <div
    class="company-card card"
    style="min-width: 30vw;"
  >
    <img
      alt="Anderson, Arias and Morrow"
      class="card-img-top"
      src="logo3.png"
      style="height: 150px; width: 200px; margin: auto;"
      width="100%"
    />
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Anderson, Arias and Morrow
      </h5>
      <h6
        class="mb-2 text-muted card-subtitle"
      >
        Size: 245 Employees
      </h6>
      <p
        class="card-text"
      >
        Somebody program how I. Face give away discussion view act inside. Your official relationship administration here.
      </p>
      <button
        class="btn btn-secondary"
        type="button"
      >
        Jobs
      </button>
    </div>
  </div>
   
</DocumentFragment>
`;
